src/eddymotion/data/dmri.py

```diff
@@ -79,6 +80,27 @@ def __len__(self):
         """Obtain the number of high-*b* orientations."""
         return self.dataobj.shape[-1]
 
+    def __eq__(self, other: Any):
+        return isinstance(other, DWI) and (
+            self.dataobj,
+            self.affine,
```

Comment:
If these are ndarrays, you're going to get issues:

```python
In [1]: import numpy as np

In [2]: (np.eye(4),) == (np.eye(4),)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[2], line 1
----> 1 (np.eye(4),) == (np.eye(4),)

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

Possibly you want something like:

```python
# Check scalars
if (self._filepath, ...) != (other._filepath):
    return False

# Check array-likes
for attr in ('dataobj', 'affine', ...):
    if not np.allclose(getattr(self, attr), getattr(other, attr)):
        return False

return True
```

Reply:
True, thanks Chris, I'll have another pass at this once I add another test dataset, or read the existing one, modify it, write it back to disk during the test, and use it as the righthand side value.

Reply:
Might be cleanest to do a round-trip test: Load from raw data inputs, serialize to HDF5, load back the HDF5 and then check equality. Then you're definitely not relying on h5py `__eq__`s hiding bugs.

Reply:
OK, will follow that path. Thanks for the insight Chris.

Reply:
Done in 8ce53d6.